src/pages/index.js

```diff
@@ -21,91 +34,155 @@ popUpImageModal.setEventListeners();
+//avatar Form - link
+const avatarFormValidator = new FormValidator(
+  constants.settings,
+  constants.avatarForm
+);
+avatarFormValidator.enableValidation();
+//card form place/Url
 const addFormValidator = new FormValidator(
   constants.settings,
   constants.addModalForm
 );
 addFormValidator.enableValidation();
-
-=======
-avatarFormValidator.enableValidation();
->>>>>>> Stashed changes
 // call queryselector on input inside form /set values
 function handleFormFill(userInfoList) {
   constants.profileTitleInput.value = userInfoList.name;
   constants.profileDescriptionInput.value = userInfoList.description;
 }
-//Buttons that Open Popup with forms
-const newUserInfo = new UserInfo(constants.selectors);
-constants.profileEditButton.addEventListener("click", () => {
-  // grab from userinfo name + descriptions
-  handleFormFill(newUserInfo.getUserInfo());
 
-  return profilePopup.openModal();
-});
 const profilePopup = new PopupWithForm(
   constants.selectors.profileSelector,
   handleProfileEditSubmit
 );
 profilePopup.setEventListeners();
+
 const cardPopUp = new PopupWithForm(
   constants.selectors.addSelector,
   handleAddModalSubmit
 );
 cardPopUp.setEventListeners();
+
+const avatarPopUp = new PopupWithForm(
+  constants.selectors.avatarSelector,
+  handleAvatarSubmit
+);
+avatarPopUp.setEventListeners();
+//Buttons that Open Popup with forms
+// grab from userinfo name + descriptions
+const newUserInfo = new UserInfo(constants.selectors);
+constants.profileEditButton.addEventListener("click", () => {
+  handleFormFill(newUserInfo.getUserInfo());
+  return profilePopup.openModal();
+});
 constants.addButton.addEventListener("click", () => {
-  addFormValidator.toggleButtonState();
   return cardPopUp.openModal();
 });
+
+constants.profileAvatar.addEventListener("click", () => {
+  return avatarPopUp.openModal();
+});
+
 //initializes new section renders inittial cards and new ones
-const cardSection = new Section(
-  {
-    items: constants.initialCards,
-    renderer: renderCard,
-  },
-  constants.selectors.cardSection
-);
-cardSection.renderItems(constants.initialCards);
+let cardSection;
+api.getInitialCards().then((data) => {
+  cardSection = new Section(
+    {
+      items: data,
+      renderer: renderCard,
+    },
+    constants.selectors.cardSection
+  );
+  cardSection.renderItems();
+});
+
 //Submit Button handler
+//Toggle Submit button State
 /**Event Handlers*/
-function handleProfileEditSubmit(modalInputs) {
-  newUserInfo.setUserInfo(modalInputs.title, modalInputs.description);
-  profilePopup.closeModal();
+function handleProfileEditSubmit(modalInputs, popUpForm) {
+  api
+    .updateInfo(modalInputs)
+    .then(() => {
+      const { title, description } = modalInputs;
+      newUserInfo.setUserInfo(title, description);
+      editFormValidator.toggleButtonState();
+    })
+    .finally(() => {
+      popUpForm.querySelector(".modal__form-button").innerText = "Save";
+      profilePopup.closeModal();
```

Comment:
`      profilePopup.closeModal();
`

You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**